cmd/main.go

```diff
@@ -202,9 +191,22 @@ func newCachedClient(ctx context.Context, cfg *rest.Config) (client.Client, erro
 
 	// populate the cache backed by shared informers that are initialized lazily on the first call
 	// for the given GVK with all resources we are interested in from the host-operator namespace
-	objectsToList := []client.ObjectList{&toolchainv1alpha1.ToolchainConfigList{}, &corev1.SecretList{}}
-	for i := range objectsToList {
-		if err := hostCluster.GetClient().List(ctx, objectsToList[i], client.InNamespace(configuration.Namespace())); err != nil {
+	objectsToList := map[string]client.ObjectList{
```

Comment:
We list only those objects we want to cache upfront. As it is mentioned in the comment above this code, the cache works in a lazy way so it starts caching GVK resources only after it is fetched for the first time. 
In other words, the cache is populated only after making the first call on the GVK resource.

If we didn't do the list here, then the very first call to the reg-service/proxy would take a lot of time because the client would start populating the cache with the resources. 
By listing the resources before the reg-service pod gets ready, we ensure that the cache is fully synced with all resources we need to have cached.

Reply:
I got that. My point is that it's easy to forget to add a new resource to this list. Let's say we are introducing some change to the proxy so it now needs to access a new resource with kind `X`. If we don't add it to this list then the proxy performance may degrade for the first request (it will degrade again after the proxy is restarted for every replica).
You can't access a new kind with adding this `X` resource to the operator role in https://github.com/codeready-toolchain/host-operator/blob/master/deploy/registration-service/registration-service.yaml#L20
So I was wondering if that yaml is a good place to add a reminder of updating the cache initialization if desired.

Reply:
As discussed in the call:
1. I'll add a comment to the Role template 
2. I'll drop the ConfigMap from the Role
3. I'll try to make the list for loop a bit more generic so it goes over all toolchain kinds except for some specific ones like TierTemplate which we don't want to cache (there can be many of them and they are huge resources)

Reply:
as for 1. and 2. see the PRs:
https://github.com/codeready-toolchain/host-operator/pull/1089
https://github.com/codeready-toolchain/registration-service/pull/463

As for the 3. item, it's easily doable to go over all kinds of the toolchain api, but there is one problem. The scheme contains api of both host & member, and at the level of the scheme we cannot decide which one is the host resource and which one belongs to the member. We could list available api groups from the cluster and exclude those kinds that are not available, but this wouldn't work in single-cluster environments.  In addition to that, host-operator SA doesn't have permission to read/list member-operator CRDs, so it would fail on authorization as well.
In other words, there are too many complications in doing it in a generic way so it's not worth it.